libmscore/cmd.cpp

```diff
@@ -2947,6 +2947,209 @@ void Score::cmdSlashRhythm()
             }
       }
 
+//---------------------------------------------------------
+//   cmdRealizeChordSymbols
+///   Realize selected chord symbols into notes on the staff.
+///
+///   If a voicing and duration type are specified, the
+///   harmony voicing settings will be overridden by the
+///   passed parameters. Otherwise, the settings set on the
+///   harmony object will be used.
+//---------------------------------------------------------
+
+void Score::cmdRealizeChordSymbols(bool literal, Voicing voicing, HDuration durationType)
+      {
+      const QList<Element*> elist = selection().elements();
+      for (Element* e : elist) {
+            if (!e->isHarmony())
+                  continue;
+            Harmony* h = toHarmony(e);
+            RealizedHarmony r = h->getRealizedHarmony();
+            Segment* seg = toSegment(h->parent());
+            Fraction duration = r.getActualDuration(durationType);
+            Fraction tick = seg->tick();
+            bool concertPitch = styleB(Sid::concertPitch);
+
+            Chord* chord = new Chord(this); //chord template
+            chord->setTrack(h->track()); //set track so notes have a track to sit on
+
+            //create chord from notes
+            QMap<int, int> notes;
+            if (voicing == Voicing::INVALID || durationType == HDuration::INVALID)
+                  notes = r.notes(); //no override, just use notes from realize harmony
+            else {
+                  //generate notes list based on overridden settings
+                  int offset = 0;
+                  Interval interval = h->staff()->part()->instrument(h->tick())->transpose();
+                  if (!concertPitch)
+                        offset = interval.chromatic;
+                  notes = r.generateNotes(h->rootTpc(), h->baseTpc(),
+                        literal, voicing, offset);
+                  }
+            QMapIterator<int, int> i(notes); //add notes to chord
+            while (i.hasNext()) {
+                  i.next();
+                  Note* note = new Note(this);
+                  NoteVal nval;
+                  nval.pitch = i.key();
+                  if (concertPitch)
+                        nval.tpc1 = i.value();
+                  else
+                        nval.tpc2 = i.value();
+                  chord->add(note); //add note first to set track and such
+                  note->setNval(nval, tick);
+                  }
+
+            setChord(seg, h->track(), chord, duration); //add chord using template
+            delete chord;
+            }
+      }
+
+//---------------------------------------------------------
+//   setChord
+//    return segment of last created chord
+//---------------------------------------------------------
+Segment* Score::setChord(Segment* segment, int track, Chord* chordTemplate, Fraction dur, Direction stemDirection)
```

Comment:
I supposed long term we could consider refactoring this with the corresponding code for notes & rests.

Reply:
That's true since corresponding code for notes works in a similar way except maybe a little less versatile and without the multiple notes and ties